ultranest/popstepsampler.py

```diff
@@ -688,7 +688,333 @@ def __next__(
+        """Return string representation."""
+        return 'PopulationSimpleSliceSampler(popsize=%d, nsteps=%d, generate_direction=%s, scale=%.g)' % (
+            self.popsize, self.nsteps, self.generate_direction, self.scale)
+
+    def region_changed(self, Ls, region):
+        """Act upon region changed. Currently unused."""
+        pass
+
+    
+    
+    def __next__(
+        self, region, Lmin, us, Ls, transform, loglike, ndraw=10,
+        plot=False, tregion=None, log=False, test=False
+    ):
+        """Sample a new live point.
+
+        Parameters
+        ----------
+        region: MLFriends object
+            Region
+        Lmin: float
+            current log-likelihood threshold
+        us: np.array((nlive, ndim))
+            live points
+        Ls: np.array(nlive)
+            loglikelihoods live points
+        transform: function
+            prior transform function
+        loglike: function
+            loglikelihood function
+        ndraw: int
+            not used
+        plot: bool
+            not used
+        tregion: bool
+            not used
+        log: bool
+            not used
+        test: bool
+            In case of test of the reversibility of the sampler, the points drawn
+            from the live points needs to be deterministic. This parameters is
+            ensuring that.
+
+        Returns
+        -------
+        u: np.array(ndim) or None
+            new point coordinates (None if not yet available)
+        p: np.array(nparams) or None
+            new point transformed coordinates (None if not yet available)
+        L: float or None
+            new point likelihood (None if not yet available)
+        nc: int
+
+        """
+        nlive, ndim = us.shape
+        
+         
+        # fill if empty:
+        if len(self.prepared_samples) == 0:
+            # choose live points
+            ilive = np.random.randint(0, nlive, size=self.popsize)
+            allu = np.array(us[ilive,:]) if not test else np.array(us)
+            allp = np.zeros((self.popsize, ndim))
+            allL = np.array(Ls[ilive])
+            nc = 0
+            n_discarded=0
+            
+                                         
+                
+            
+            interval_final=0. 
+            
+            for k in range(self.nsteps):
+                # Defining scale jitter
+                factor_scale=self.scale_jitter_func()
+                # Defining slice direction
+                v = self.generate_direction(allu, region,scale= 1.0)*self.scale*factor_scale
+                 
+                
+                # limite of the slice based on the unit cube boundaries
+                tleft_unitcube,tright_unitcube= unitcube_line_intersection(allu, v)
+                # Defining bound of the slice
+                tleft_worker,tright_worker = self.slice_limit(tleft_unitcube,tright_unitcube)
+                tleft,tright = self.slice_limit(tleft_unitcube,tright_unitcube)
```

Comment:
these two slice_limit lines need a comment what is happening here, because it is not obvious.